File: app/Quotes/ocaml/src/core/change_pass.ml
(* Copyright 09-07-2018 ºDeme
   GNU General Public License - V3 <http://www.gnu.org/licenses/>
*)

let process cgi rq = Json.(Cgi.(
    change_pass cgi
      (rrq rq "user" rstring)
      (rrq rq "oldPass" rstring)
      (rrq rq "newPass" rstring)
  ))

File: app/HDoc/h/src/ChangePass.hs
-- Copyright 18-Feb-2020 ºDeme
-- GNU General Public License - V3 <http://www.gnu.org/licenses/>

--- Change passord source.

module ChangePass where

import qualified Dm.Cgi as Cgi
import qualified Dm.Js as Js
import qualified Dm.Map as Map

process :: Cgi.T -> Map.T Js.T -> IO (String)
process cgi rq = do
  user <- Cgi.rrq (Cgi.err "") rq "user" Js.rs
  old <- Cgi.rrq (Cgi.err "") rq "old" Js.rs
  new <- Cgi.rrq (Cgi.err "") rq "new" Js.rs
  Cgi.changePass cgi user old new
